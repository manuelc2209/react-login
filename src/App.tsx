import React, { useState } from 'react';
import styled from 'styled-components';
import './index.css';
import { Routes, Route } from 'react-router';
import { useNavigate } from 'react-router-dom';

import data from './data/index.json';
import { COLOR_PRIMARY_2 } from './GlobalStyles';
import {
    Button,
    DashboardUI,
    Header,
    LoginUI,
    Portefolio,
    RegisterUI,
    ProjectsUI,
    Skills
} from './components';

const StyledContainer = styled.div`
    height: inherit;
    display: flex;
    flex-direction: column;
`;

const StyledContent = styled.div`
    display: flex;
    flex-direction: row;
    background-image: url('https://64.media.tumblr.com/2df67fe7bdbba84c88cdbbdf84fd2743/tumblr_nqgvxz92HC1s85u2fo1_500.gif');
    background-position: center;
    background-repeat: no-repeat;
    background-size: cover;
    position: relative;

    @media (max-width: 450px) {
        flex-direction: column;
        flex: 1;
    }

    @media (min-width: 900px) {
        height: 100%;
    }
`;

const StyledOverlay = styled.div`
    display: flex;
    flex: 1;
    background-color: #20202085;

    @media (max-width: 900px) {
        flex-direction: column;
    }
`;

const StyledColumn = styled.div`
    width: 50%;
    height: 100%;
    display: flex;
    flex-direction: column;
    flex: 1;

    @media (max-width: 450px) {
        width: 100%;
    }

    @media (max-width: 900px) {
        width: 100%;
    }
`;

const StyledHeader = styled(Header)`
    border-bottom: 1px solid ${COLOR_PRIMARY_2};
    display: flex;
    flex-direction: row;
    justify-content: flex-end;
    height: 60px;
    align-items: center;

    > * {
        margin: 10px;
    }
`;

export const App: React.FC = () => {
    const [disabled, setDisabled] = useState(false);
    const [login, setLogin] = useState(false);
    const [register, setRegister] = useState(false);
    const navigate = useNavigate();
    const { profile, skills } = data;

    function resetState() {
        setRegister(false);
        setLogin(false);
    }

    const handleOnClick = () => {
        setDisabled(true);
        // Fake timeout to simmulate a request while api is not ready
        setTimeout(() => setDisabled(false), 2500);
    };

    const handleProjects = () => {
        resetState();
        navigate('/projects');
    };

    const handleLogin = () => {
        setLogin(true);
        navigate(login ? '/' : '/login');
    };

    const handleBack = () => {
        setLogin(false);
        setRegister(false);
        navigate('/');
    };

    const handleRegister = () => {
        setRegister(true);
        navigate(register ? '/' : '/register');
    };

    // REFACTOR THIS INTO UI

    const App = (
        <StyledContainer>
            <StyledHeader>
                {login ? (
                    <Button size="large" label="Log out" onClick={() => handleBack && handleBack()} />
                ) : (
                    <>
                        <Button size="large" label="Log in" onClick={() => handleLogin && handleLogin()} />
                        <Button
                            size="large"
                            label="Sign up"
                            onClick={() => handleRegister && handleRegister()}
                        />
                        <Button
                            size="large"
                            label="Projects"
                            onClick={() => handleProjects && handleProjects()}
                        />
                    </>
                )}
            </StyledHeader>
            <StyledContent>
                <StyledOverlay>
                    <StyledColumn>
                        <Portefolio
                            label={profile.label}
                            img={profile.img}
                            headline={profile.headline}
                            link={profile.link}
                            linkLabel={profile.linkLabel}
                            cards={profile.cards}
                        />
                    </StyledColumn>
                    <StyledColumn>
                        <Skills skills={skills} />
                    </StyledColumn>
                </StyledOverlay>
            </StyledContent>
        </StyledContainer>
    );

    const loginButtonProps = {
        label: 'Go back',
        size: 'large',
        disabled: false,
        onClick: () => handleBack()
    };

    const loginFormProps = {
        nameLabel: 'Username:',
        nicknamePlaceholder: 'Please type in your Nickname here:',
        passwordLabel: 'Password:',
        passwordPlaceholder: 'Please type in your password here:',
        disabled: false,
        onClick: () => handleOnClick()
    };

    const Login = <LoginUI buttonProperties={loginButtonProps} formProperties={loginFormProps} />;

    const registerButtonProps = {
        label: 'Go back',
        size: 'large',
        disabled: false,
        onClick: () => handleBack()
    };

    const formProps = {
        name: 'Username',
        disabled: false,
        validatePassword: true,
        doubleValidation: true,
        passwordLabel: 'Password:',
        nicknamePlaceholder: 'Username Placeholder',
        passwordPlaceholder: 'Password Placeholder',
        repeatPasswordPlaceholder: 'Repeat your password here',
        onClick: () => handleOnClick()
    };

    const Register = <RegisterUI buttonProperties={registerButtonProps} formProperties={formProps} />;
    const Dashboard = <DashboardUI />;

    const buttonProps = {
        label: 'Go back',
        size: 'large',
        disabled: false,
        onClick: () => handleBack()
    };

    const Projects = <ProjectsUI buttonProperties={buttonProps} />;

    return (
        <Routes>
            <Route path="/" element={App} />
            <Route path="register" element={Register} />
            <Route path="login" element={Login} />
            <Route path="projects" element={Projects} />
            <Route path="dashboard" element={Dashboard} />
            <Route path="*" element={App} />
        </Routes>
    );
};
