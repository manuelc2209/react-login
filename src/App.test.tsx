import React from 'react';
import ReactDOM from 'react-dom';
import { BrowserRouter } from 'react-router-dom';
import renderer from 'react-test-renderer';

import { App } from './App';

const AppComponent = (
    <BrowserRouter>
        <App />
    </BrowserRouter>
);

it('renders snapshot correctly', () => {
    const tree = renderer.create(AppComponent).toJSON();
    expect(tree).toMatchSnapshot();
});

it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(AppComponent, div);
    ReactDOM.unmountComponentAtNode(div);
});
