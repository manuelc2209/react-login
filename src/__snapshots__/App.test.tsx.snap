// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders snapshot correctly 1`] = `
<div
  className="sc-hOGkXu cwOgHW"
>
  <div
    className="sc-pVTFL fZkDsV sc-jQrDum fzjBSh"
  >
    <button
      className="sc-gsDKAQ dWwrqJ"
      disabled={false}
      onClick={[Function]}
      onMouseDown={[Function]}
      onMouseUp={[Function]}
      size="large"
    >
      <span
        className="sc-bdvvtL bBtbgR"
      >
        Log in
      </span>
    </button>
    <button
      className="sc-gsDKAQ dWwrqJ"
      disabled={false}
      onClick={[Function]}
      onMouseDown={[Function]}
      onMouseUp={[Function]}
      size="large"
    >
      <span
        className="sc-bdvvtL bBtbgR"
      >
        Sign up
      </span>
    </button>
    <button
      className="sc-gsDKAQ dWwrqJ"
      disabled={false}
      onClick={[Function]}
      onMouseDown={[Function]}
      onMouseUp={[Function]}
      size="large"
    >
      <span
        className="sc-bdvvtL bBtbgR"
      >
        Projects
      </span>
    </button>
  </div>
  <div
    className="sc-dtMgUX iDIaip"
  >
    <div
      className="sc-cZMNgc cJlINy"
    >
      <div
        className="sc-jUosCB kRzrff"
      >
        <div
          className="sc-jJoQJp kvWGrA"
        >
          <img
            className="sc-llYSUQ cNUGOY"
            src="https://media-exp1.licdn.com/dms/image/C4D03AQHnmnhLoeWeUQ/profile-displayphoto-shrink_800_800/0/1643239289504?e=1649289600&v=beta&t=Tch_XNfZXK3s4ZM3tdSAyDXkErE3wh9kTf3m4dSfMuw"
          />
          <span
            className="sc-iwjdpV inWMhS"
          >
            Who am I?
          </span>
          <div
            className="sc-giYglK bSujg"
          >
            <span
              className="sc-cxpSdN MACTq"
            >
              I'm a Front-End Developer for 
              <a
                className="sc-iJKOTD"
                href="https://www.cleverti.com/"
              >
                Cleverti
              </a>
            </span>
          </div>
          <div
            className="sc-ezbkAF jOWQzN"
          >
            <div
              className="sc-dkPtRN eGWjTe sc-bYoBSM lbEZDW"
            >
              <a
                className="sc-jRQBWg"
                href="https://www.linkedin.com/in/manuel-correia2209/"
                target="_self"
              >
                <div
                  className="sc-hKwDye"
                >
                  <img
                    className="sc-gKclnd huNaHt"
                    src="https://hermes.digitalinnovation.one/articles/cover/669b63fe-912d-484c-b313-95a46a8ee25e.png"
                  />
                </div>
              </a>
              <div
                className="sc-eCImPb dlpleo"
              >
                <span
                  className="sc-iCfMLu jSHYWD"
                >
                  LinkedIn
                </span>
                
              </div>
            </div>
            <div
              className="sc-dkPtRN eGWjTe sc-bYoBSM lbEZDW"
            >
              <a
                className="sc-jRQBWg"
                href="https://github.com/manuelc2209/react-ui"
                target="_self"
              >
                <div
                  className="sc-hKwDye"
                >
                  <img
                    className="sc-gKclnd huNaHt"
                    src="https://hermes.digitalinnovation.one/articles/cover/7b4fbb1d-f57a-4420-9209-ee976b171bdf.jpg"
                  />
                </div>
              </a>
              <div
                className="sc-eCImPb dlpleo"
              >
                <span
                  className="sc-iCfMLu jSHYWD"
                >
                  Github
                </span>
                
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        className="sc-jUosCB kRzrff"
      >
        <div
          className="sc-cCcXHH wGCrs"
        >
          <span
            className="sc-gWXbKe wIHqs"
          >
            My skills:
          </span>
          <div
            className="sc-hiCibw iURpTG"
            title="90"
          >
            <span
              className="sc-gWXbKe wIHqs"
            />
            <span
              className="sc-gWXbKe wIHqs"
            >
              React
            </span>
          </div>
          <div
            className="sc-hiCibw iURpTG"
            title="90"
          >
            <span
              className="sc-gWXbKe wIHqs"
            />
            <span
              className="sc-gWXbKe wIHqs"
            >
              Angular
            </span>
          </div>
          <div
            className="sc-hiCibw iURpTG"
            title="90"
          >
            <span
              className="sc-gWXbKe wIHqs"
            />
            <span
              className="sc-gWXbKe wIHqs"
            >
              Styled-Components
            </span>
          </div>
          <div
            className="sc-hiCibw cKjgBy"
            title="85"
          >
            <span
              className="sc-gWXbKe wIHqs"
            />
            <span
              className="sc-gWXbKe wIHqs"
            >
              React-testing
            </span>
          </div>
          <div
            className="sc-hiCibw blTjcC"
            title="80"
          >
            <span
              className="sc-gWXbKe wIHqs"
            />
            <span
              className="sc-gWXbKe wIHqs"
            >
              Jest
            </span>
          </div>
          <div
            className="sc-hiCibw blTjcC"
            title="80"
          >
            <span
              className="sc-gWXbKe wIHqs"
            />
            <span
              className="sc-gWXbKe wIHqs"
            >
              Git
            </span>
          </div>
          <div
            className="sc-hiCibw dmbkfW"
            title="75"
          >
            <span
              className="sc-gWXbKe wIHqs"
            />
            <span
              className="sc-gWXbKe wIHqs"
            >
              TypeScript
            </span>
          </div>
          <div
            className="sc-hiCibw gHkpzq"
            title="60"
          >
            <span
              className="sc-gWXbKe wIHqs"
            />
            <span
              className="sc-gWXbKe wIHqs"
            >
              Scss
            </span>
          </div>
          <div
            className="sc-hiCibw bkNlIC"
            title="50"
          >
            <span
              className="sc-gWXbKe wIHqs"
            />
            <span
              className="sc-gWXbKe wIHqs"
            >
              Css
            </span>
          </div>
          <div
            className="sc-hiCibw bkNlIC"
            title="50"
          >
            <span
              className="sc-gWXbKe wIHqs"
            />
            <span
              className="sc-gWXbKe wIHqs"
            >
              Vue
            </span>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
